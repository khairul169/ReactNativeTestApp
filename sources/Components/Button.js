import React from "react";
import { View, TouchableOpacity, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

const Button = props => {
  return (
    <View
      style={[styles.container, props.full && styles.fullButton, props.style]}
    >
      <TouchableOpacity style={styles.button} onPress={props.onPress}>
        <Text style={[styles.text, props.titleStyle]}>{props.title}</Text>
      </TouchableOpacity>
    </View>
  );
};

Button.propTypes = {
  title: PropTypes.string.isRequired,
  full: PropTypes.bool,
  style: PropTypes.object,
  titleStyle: PropTypes.object,
  onPress: PropTypes.func
};

Button.defaultProp = {
  full: false
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    elevation: 3,
    alignSelf: "center",
    marginVertical: 12
  },
  fullButton: {
    alignSelf: "stretch"
  },
  button: {
    paddingHorizontal: 32,
    paddingVertical: 8,
    alignSelf: "stretch",
    alignItems: "center"
  },
  text: {
    fontSize: 16,
    color: "#333",
    fontWeight: "100"
  }
});

export default Button;
