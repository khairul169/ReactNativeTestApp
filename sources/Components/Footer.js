import React from "react";
import { View, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

const Footer = props => {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>{props.text}</Text>
    </View>
  );
};

Footer.propTypes = {
  text: PropTypes.string.isRequired
};

Footer.defaultProps = {
  text: "This is a footer."
};

const styles = StyleSheet.create({
  container: {
    height: 48,
    alignItems: "center",
    justifyContent: "center",
    borderTopColor: "#eee",
    borderTopWidth: 1
  },
  text: {
    color: "#333",
    fontSize: 14
  }
});

export default Footer;
