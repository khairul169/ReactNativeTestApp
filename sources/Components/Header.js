import React from "react";
import { View, Text, StyleSheet } from "react-native";
import PropTypes from "prop-types";

const Header = props => {
  return (
    <View style={styles.container}>
      <Text style={styles.titleText}>{props.title}</Text>
    </View>
  );
};

Header.propTypes = {
  title: PropTypes.string
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#fff",
    elevation: 3,
    flexDirection: "row",
    alignItems: "stretch",
    height: 48
  },
  titleText: {
    color: "#333",
    fontSize: 18,
    fontWeight: "100",
    alignSelf: "center",
    marginLeft: 24
  }
});

export default Header;
