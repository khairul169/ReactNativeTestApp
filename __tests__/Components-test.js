import React from "react";
import renderer from "react-test-renderer";

// Components
import Header from "../sources/Components/Header";
import Button from "../sources/Components/Button";
import Footer from "../sources/Components/Footer";

describe("<Header />", () => {
  const defaultProps = {
    title: "Header"
  };
  const wrapper = renderer.create(<Header {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<Button />", () => {
  const defaultProps = {
    title: "Button"
  };
  const wrapper = renderer.create(<Button {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("<Footer />", () => {
  const defaultProps = {};
  const wrapper = renderer.create(<Footer {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
