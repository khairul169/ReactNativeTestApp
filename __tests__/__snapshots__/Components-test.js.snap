// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "center",
        "backgroundColor": "#fff",
        "elevation": 3,
        "marginVertical": 12,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    accessible={true}
    clickable={false}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "opacity": 1,
        "paddingHorizontal": 32,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333",
            "fontSize": 16,
            "fontWeight": "100",
          },
          undefined,
        ]
      }
    >
      Button
    </Text>
  </View>
</View>
`;

exports[`<Footer /> render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderTopColor": "#eee",
      "borderTopWidth": 1,
      "height": 48,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#333",
        "fontSize": 14,
      }
    }
  >
    This is a footer.
  </Text>
</View>
`;

exports[`<Header /> render 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#fff",
      "elevation": 3,
      "flexDirection": "row",
      "height": 48,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "center",
        "color": "#333",
        "fontSize": 18,
        "fontWeight": "100",
        "marginLeft": 24,
      }
    }
  >
    Header
  </Text>
</View>
`;
